packages/grub-config/config/grub.cfg

```diff
@@ -29,8 +29,13 @@ menuentry "cOS (fallback)" --id fallback {
 }
 
 menuentry "cOS recovery" --id recovery {
+  if search.file /cOS/recovery.squashfs ; then
```

Comment:
I'm not very enthusiastic about this. If no squashfs is found GRUB will display a message that failed to find it which is very confusing for the user, but it works as expected as it boots correctly into recovery. I couldn't find a way to make it "less verbose" and avoid the misleading error message.

Reply:
yup, that's why I had to end up with a simple if that seems to work just fine: https://github.com/rancher-sandbox/cOS-toolkit/pull/241/files#diff-82bb6cdc894381fa9e4ed0ad09b4bad8d0cbd30079b88abf7aa379d5f9ab881fR2 . the trick part is that each image that is intended to be used as recovery and/or as a bootable system have to deal with the case when it's being used for recovery or not